sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -474,6 +474,20 @@ case class MapObjects private(
           s"$seq == null ? $array[$loopIndex] : $seq.apply($loopIndex)"
     }
 
+    // Make a copy of the data if it's unsafe-backed
+    val genFunctionValue = lambdaFunction.dataType match {
```

Comment:
we're calling `toString` on a `UTF8String`, so maybe there's no need to clone `UTF8String`s?

```java
...
/* 072 */    value8 = MapObjects_loopValue2.getUTF8String(0);
...
/* 082 */    funcResult = value8.toString();
...
/* 086 */    value7 = (java.lang.String) funcResult;
...
/* 128 */    convertedArray[loopIndex] = ...;
```